### PR TITLE
lint devtools, clean-up

### PR DESCRIPTION
**Pull request by dckc**
I'd like us to split #10 into some smaller PRs. This is a candidate for a 1st one.

 - [ ] deal with remaining errors
 - [ ] TODO around `test` / `anyTest`
 - [ ] `yarn lint` in ci

### CODE REVIEW

File: contract/package.json [hunk 1]
```diff
@@ -44,20 +44,20 @@
...
-    "eslint-plugin-prettier": "^5.0.0",
+    "eslint-plugin-import": "^2.29.1",
+    "eslint-plugin-jsdoc": "^48.5.2",
+    "eslint-plugin-prettier": "^5.1.3",
     "import-meta-resolve": "^2.2.1",
     "prettier-plugin-jsdoc": "^1.0.0",
     "rollup": "^4.18.0",
     "type-coverage": "^2.26.3",
-    "typescript": "~5.2.2"
+    "typescript": "^5.5.3"
```
Comment (dckc) [inline contract/package.json:59]: @turadg I had copied versions from agoric-sdk into `package.json`; did you update `typescript` to "current from npm"?

Did you use any `yarn upgrade ...` tooling? or just edit `package.json`?

File: contract/src/platform-goals/start-contract.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ export const installContract = async (
  *   startArgs?: StartArgs;
  *   issuerNames?: string[];
  * }} opts
+ * @param {object} [privateArgs]
```
Comment (dckc) [inline contract/src/platform-goals/start-contract.js:41]: I tried to get rid of this copy of `privateArgs`. There's already a `privateArgs` in `startArgs`.

File: contract/package.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
     "@endo/eslint-plugin": "^0.5.2",
     "@endo/nat": "^4.1.27",
     "@endo/promise-kit": "^0.2.56",
-    "@endo/ses-ava": "^1.4.0",
+    "@endo/ses-ava": "^1.2.4",
```
Comment (dckc) [inline contract/package.json:35]: this bumps ses-ava down/back, right? I'm interested to know why.

File: contract/package.json [hunk 4]
```diff
@@ -30,7 +30,7 @@
     "@agoric/zone": "^0.2.2",
     "@cosmjs/amino": "^0.32.3",
     "@cosmjs/proto-signing": "^0.32.3",
-    "@endo/eslint-plugin": "^0.5.2",
+    "@endo/eslint-plugin": "^2.2.0",
```
Comment (dckc) [inline contract/package.json:32]: I'm still learning... how did you choose that version number? is it just the current in npm?

Did anything in particular prompt changing it? Should we generally recommend using the then-current versions of all lint-plugins in https://github.com/Agoric/agoric-sdk/discussions/8274 ?

File: contract/package.json [hunk 5]
```diff
@@ -14,8 +14,7 @@
     "build": "yarn build:deployer",
     "build:deployer": "rollup -c rollup.config.mjs",
     "test": "ava",
-    "lint": "eslint '**/*.js'",
-    "lint:types": "tsc -p jsconfig.json",
```
Comment (dckc) [inline contract/package.json:18]: we don't do `lint:types` any more?

File: contract/src/orca.proposal.js [hunk 6]
```diff
@@ -1,29 +1,21 @@
 /* eslint-disable -- FIXME */
-// @ts-check
-import { allValues } from './objectTools.js';
+// @ts-nocheck -- FIXME
```
Comment (dckc) [inline contract/src/orca.proposal.js:2]: this fixes some type stuff but then punts on type checking?

File: contract/package.json (hunk 1 above)
Comment (turadg) [reply contract/package.json:59]: I think this is your change. I don't recall changing this from your branch. It's [the version in agoric-sdk](https://github.com/Agoric/agoric-sdk/blob/master/package.json#L37)

File: contract/package.json (hunk 5 above)
Comment (turadg) [reply contract/package.json:18]: nothing was running `lint:types`. agoric-sdk had it that way for backwards compatibility. That scheme requires something like `run-s` to lint all the `lint:foo` targets.

When you want to lint types, I think `yarn tsc` is handy enough. Fewer chars. And it's fast enough that when you want to eslint, it doesn't hurt to wait for tsc first

File: contract/package.json (hunk 4 above)
Comment (turadg) [reply contract/package.json:32]: for this one I used `yarn upgrade-interactive`

File: contract/tsconfig.json [hunk 7]
```diff
@@ -1,16 +1,19 @@
...
-    "module": "ESNext",
-    "moduleResolution": "node",
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
     "lib": ["ESNext", "DOM"],
-    "strict": true,
+    "checkJs": true,
+    "strict": false,
+    "noImplicitAny": false,
+    "useUnknownInCatchVariables": false,
```
Comment (dckc) [inline contract/tsconfig.json:10]: this looks like some hard-won knowledge.

Something to add to https://github.com/Agoric/agoric-sdk/discussions/8274 ?

File: contract/package.json (hunk 3 above)
Comment (turadg) [reply contract/package.json:35]: 1.2.4 is actually the latest version of https://www.npmjs.com/package/@endo/ses-ava

File: contract/src/orca.proposal.js (hunk 6 above)
Comment (turadg) [reply contract/src/orca.proposal.js:2]: yeah I timed out

File: contract/src/platform-goals/start-contract.js (hunk 2 above)
Comment (turadg) [reply contract/src/platform-goals/start-contract.js:41]: oic. I'll revise

File: contract/tsconfig.json (hunk 7 above)
Comment (turadg) [reply contract/tsconfig.json:10]: they manifest pretty loudly. `useUnknownInCatchVariables` and `noImplicitAny` were relaxing `strict: true` but there were so many other hurdles I just turned that off (it's off in sdk). I'll remove these two redundant flags

Comment (turadg) [reply contract/tsconfig.json]: oh, most of the other problems are solved so I'll flip it to true and suppress a few

**Review by turadg — APPROVED** [review]
LGTM ;-)

**Pull request merged**